src/main/scala/Monday.scala

```diff
@@ -1,144 +1,160 @@
+      if (combination.substring(i, i + 1) == c.toString) {
+        myreturn = myreturn + d
 
+      }
+      else {
+        myreturn += combination.substring(i, i + 1)
+      }
+
+    }
+    myreturn
   }
-  return myreturn
-}
 
-println(strings2("Ha","llo",'a','e'))
+  println(strings2("Ha", "llo", 'a', 'e'))
 
-def add(a:Int, b:Int, c:Boolean):Int={
-  if(a==0||b==0){
-    if(b>a){
-      return b
+  def conditionals(a: Int, b: Int, bool: Boolean): Int = {
+    if (a == 0 || b == 0) {
+      if (b > a) {
+        b
+      }
+      else {
+        a
+      }
     }
-    else{
-      return a
+    else if (bool) {
+      a + b
+    }
+    else {
+      a * b
     }
   }
-  else if(c){
-    return a+b
-  }
-  else{
-    return a*b
-  }
-}
-println(add(0,3,3!=3))
 
-def iteration(a:String,b:Int){
-  for(i<- 1 to b){
-    println(a)
-  }
-}
+  println(conditionals(0, 3, 3 != 3))
 
-iteration("H",4)
+  def iteration(a: String, b: Int) {
+    for (i <- 1 to b) {
+      println(a)
+    }
+  }
 
+  iteration("H", 4)
 
 
-def iteration2(a:String,b:Int){
-  for(i<- 1 to b){
-    for(i<- 1 to b){
-      print(a)
+  def iteration2(a: String, b: Int) {
+    for (i <- 1 to b) {
+      for (i <- 1 to b) {
+        print(a)
+      }
+      print("\n")
     }
-    print("\n")
   }
-}
 
-iteration2("H",4)
+  iteration2("H", 4)
 
-def iteration3(a:String,b:String,c:Int){
-  for(i<- 1 to c){
-    if(i%3==0 && i%5==0){
-      print(a+b)
-    }
-    else if(i%3==0){
-      print(a)
-    }
-    else if(i%5==0){
-      print(b)
-    }
-    else{
-      print(i)
+  def iteration3(a: String, b: String, c: Int) {
+    for (i <- 1 to c) {
+      if (i % 3 == 0 && i % 5 == 0) {
+        print(a + b)
+      }
+      else if (i % 3 == 0) {
+        print(a)
+      }
+      else if (i % 5 == 0) {
+        print(b)
+      }
+      else {
+        print(i)
+      }
+      if (i != c) {
+        print(",")
+      }
     }
-    if(i==c){
+  }
 
-    }
-    else{
-      print(",")
+  iteration3("Fizz", "Buzz", 15)
+  println("")
+
+  def recursion(a: String, b: Int) {
+    if (b > 0) {
+      println(a)
+      recursion(a, b - 1)
     }
   }
-}
-iteration3("Fizz","Buzz",15)
 
+  recursion("H", 4)
 
-def recursion(a:String,b:Int){
-  if(b>0){
-    println(a)
-    recursion(a,b-1)
-  }
-  else{
-    println(a)
-  }
+  def recursion2(a: String, b: Int) {
 
-}
+    val columns = b
+    if (b > 0) {
+      println(a*b)
+      recursion2(a, b - 1)
 
-recursion("H",4)
+    }
+    else {
+    }
+  }
 
-def recursion2(a:String,b:Int){
-  if(b>0){
-    def nestedrecursion(c:String,d:Int){
-      if(d>0){
-        print(c)
-        //nestedrecursion(c,d-1)
-      }
-      else{
-        println("")
-        nestedrecursion(a,b)
+  recursion2("H", 4) //needs work, reduces the amount of times it prints the string with each turn
+
+  def recursion3(a: String, b: String, c: Int): String = {
+    var output = ""
+    logicChecks(a, b, c)
+
+    def logicChecks(a: String, b: String, c: Int) {
```

Comment:
names

output = a + b + output could be done as output += a+b

would like to see this as pattern matching rather than ifs